react/react2/week1/github-user-searcher/src/search-component/SearchUsersList.js

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useEffect, useState } from "react";
+import { usersContext } from "./SearchMain";
+import { SearchUsersItem } from "./SearchUsersItem";
+
+export const ApiTotalData = React.createContext();
+
+export const SearchUsersList = () => {
+  const [apiTotalData, setApiTotalData] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  // take data from SearchMain input
+  const query = useContext(usersContext);
+
+  const api = `https://api.github.com/search/users?q=${query}`;
+
+  useEffect(() => {
+    if (!query) {
+      setApiTotalData([]);
+      return;
+    }
+    setLoading(true);
+    (async () => {
+      try {
+        const response = await (await fetch(api)).json();
```

Comment:
const fecthData = async () => {
    try {
      const fetchApi = await fetch(api);
      const apiData = await fetchApi.json();
      const result = apiData.items.map((data) => data.login);
      setApiTotalData(result);
    } catch (error) {
      console.log(error);
    } finally {
      setLoading(false);
    }
  };
  useEffect(() => {
    if (!query) {
      setApiTotalData([]);
      return;
    }
    setLoading(true);
    fecthData();
  }, [api, query]);

Reply:
Hi, For this part. if I put the fecthData function outside of useEffect , it gave me an error. when I put the fecthData function inside useEffect, It will not give me an error. I do not know why? would you pls tell me??

Reply:
it should be inside the useEffect, what I was trying to say is to separate this function into parts  `const response = await (await fetch(api)).json(); ` as you did here 
```
const fetchApi = await fetch(api);
const apiData = await fetchApi.json();
```

and try to catch the error if the response of fetchApi is not 200, as you can see on the link I sent you

Reply:
HI, I mean here why can not put that function outside useEffect? I know inside the useEffect is working.